pkg/utils/retry.go

```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"time"
+)
+
+// Retry will make a given number of attempts to run the provided function
+// with exponential backoff
+func Retry(attempts int, sleep time.Duration, fn func() error) error {
```

Comment:
Generally (not sure in go) functions that take function pointers like this ask for the function first, then the args.